worlds/sa2b/Missions.py

```diff
@@ -193,10 +194,10 @@
     "Cannon's Core - ",
 ]
 
-def get_mission_count_table(multiworld: MultiWorld, player: int):
+def get_mission_count_table(multiworld: MultiWorld, world: World, player: int):
     mission_count_table: typing.Dict[int, int] = {}
 
-    if multiworld.goal[player] == 3:
+    if world.options.goal == 3:
```

Comment:
```suggestion
    if world.options.goal == "grand_prix":
```

Are you aware you can write this? It's a bit easier to follow when one does not know that 3 happens to be grand prix.

Reply:
Alternatively, could `from .options import Goal` and then do `world.options.goal == Goal.option_grand_prix` if you want the typing which is how I prefer it.